lib/SILOptimizer/Transforms/AllocBoxToStack.cpp

```diff
@@ -922,9 +1034,11 @@ class AllocBoxToStack : public SILFunctionTransform {
     AllocBoxToStackState pass(this);
     for (auto &BB : *getFunction()) {
       for (auto &I : BB)
-        if (auto *ABI = dyn_cast<AllocBoxInst>(&I))
-          if (canPromoteAllocBox(ABI, pass.PromotedOperands))
+        if (auto *ABI = dyn_cast<AllocBoxInst>(&I)) {
+          std::unordered_set<SILFunction*> VisitedCallees;
```

Comment:
Please don't use std::unordered_set. Instead, use a SmallPointerSet or something like that. I would also move this out of the main inner loop and just clear it as you go around to prevent reallocation.